File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {

  public static List<Integer>[] adj = new ArrayList[1002];
  public static boolean[] vis = new boolean[1002];

  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st = null;

    st = new StringTokenizer(br.readLine());
    int n = Integer.parseInt(st.nextToken());
    int m = Integer.parseInt(st.nextToken());

    for (int i = 1; i <= n; i++) {
      adj[i] = new ArrayList<>();
    }

    while(m-- > 0){
      st = new StringTokenizer(br.readLine());
      int u = Integer.parseInt(st.nextToken());
      int v = Integer.parseInt(st.nextToken());

      adj[u].add(v);
      adj[v].add(u);
    }

    int count = 0;
    for(int i=1; i<=n; i++){
      if(vis[i]) continue;
      vis[i] = true;
      Queue<Integer> Q = new LinkedList<>();
      Q.add(i);
      count++;

      while(!Q.isEmpty()){
        int cur = Q.peek(); Q.poll();
        for(Integer nxt:adj[cur]){
          if(vis[nxt]) continue;

          vis[nxt] = true;
          Q.add(nxt);
        }
      }
    }
    System.out.println(count);
  }
}

File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.cc
#include <bits/stdc++.h>
using namespace std;

vector<int> adj[1002];
bool vis[1002];

int main(){
  ios::sync_with_stdio(0);
  cin.tie(0);

  int n, m;
  cin >> n >> m;

  while(m--){
    int u, v;
    cin >> u >> v;

    adj[u].push_back(v);
    adj[v].push_back(u);
  }

  queue<int> q;
  int cnt = 0;
  for(int i=1; i<=n; i++){
    if(vis[i]) continue;

    cnt++;
    q.push(i);
    while(!q.empty()){
      int cur = q.front(); q.pop();

      for(int nxt:adj[cur]){
        if(vis[nxt]) continue;

        vis[nxt] = true;
        q.push(nxt);
      }
    }
  }

  cout << cnt;
}